### PR TITLE
[COASTAL-1411] Add CustomAlert with image support

### PR DESCRIPTION
**Pull request by Camji55**
https://tidepool.atlassian.net/browse/COASTAL-1411

### CODE REVIEW

File: LoopKitUI/Views/CustomAlert.swift [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    
+    public class CustomAlertPresentingViewController: UIViewController {
+        private var alertController: UIAlertController?
+        
+        private let alertBinding: Binding<CustomAlert?>
+        private let guidanceColors: GuidanceColors
+        
+        fileprivate init(alertBinding: Binding<CustomAlert?>, guidanceColors: GuidanceColors) {
+            self.alertBinding = alertBinding
+            self.guidanceColors = guidanceColors
```
Comment (nhamming) [inline LoopKitUI/Views/CustomAlert.swift:53]: not blocking but curious why the `guidanceColors` is both pulled from the environment and passed in for initialization?

**Review by nhamming — APPROVED** [review]
LGTM

**Review by ps2 — COMMENTED** [review]
I would prefer we add actions and images to the Loop alert system, rather than creating a new kind of alert. Supporting actions within iOS notifications is really a nice feature that we've talked about for a long time; it can allow users to take actions on alerts from the watch, or lock screen, which can be pretty convenient.

Also, "Custom" is a pretty vague modifier for a class. What particular customization is being provided?  The words used in answering that might provide input for a more specific name. Like if we're customizing how actions are done, then ActionableAlert, or something might be more appropriate, for example.

**Comment by ps2** [comment]
I also don't see code that is showing how this is used; that's important to evaluate this kind of change.

File: LoopKitUI/Views/CustomAlert.swift (hunk 1 above)
Comment (Camji55) [reply LoopKitUI/Views/CustomAlert.swift:53]: I don't think `guidanceColors` gets picked up by a `UIViewController` just the wrapping `UIViewControllerRepresentable` which is a SwiftUI view. I can test this though.

**Pull request closed without merging**